app/lib/Parsers/ExpressionParser/ExpressionVisitor.php

```diff
@@ -108,6 +108,11 @@ protected function initializeFunctions() {
 			'fromUnixtime'	=> xcallable(function($arg) { return date('c', $arg); }),
 			'existsInList'	=> xcallable(function($arg1, $arg2) { return caItemExists($arg1, $arg2); })
 		);
+
+		// Let application plugins add their own expression functions
```

Comment:
> Feedback from original MR [#6 (comment)](https://github.com/gaiaresources/providence/pull/6#discussion_r1239351016)
> 
> > We should probably have an allow list of these in config (app.conf) somewhere cos this is a pretty big security hole thinking about it.

@kehh what if we added a hook here? made the security implications the plugins problem? (will add link here for my suggestions inside the plugin)